packages/nextcloud/lib/src/webdav/path_uri.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:collection/collection.dart';
+import 'package:meta/meta.dart';
+
+/// A [Uri] that only exposes the path handling.
+@immutable
+class PathUri {
```

Comment:
Is there a reason why this does not implement the `Uri` interface?

Reply:
Then we would need to expose everything else as well. I thought one of the reasons for this was that we want to hide parts of Uri that we don't want to be exposed?